tests/schema/test_settings_validator.py

```diff
@@ -0,0 +1,103 @@
+from math import isclose
+from typing import Any, Dict
+
+import avro
+import pytest
+
+from gsy_framework.exceptions import GSySerializationException
+from gsy_framework.schema.validators import AVROSimulationSettingsSerializer
+
+
+class TestSimulationSettingsValidator:
+    # pylint: disable=attribute-defined-outside-init
+
+    def setup_method(self):
+        self._data = {
+            "duration": timedelta(days=7),
+            "slot_length": timedelta(seconds=900),
+            "tick_length": timedelta(seconds=15),
+            "market_count": 1,
+            "cloud_coverage": 0,
+            "pv_user_profile": None,
+            "market_maker_rate": 30,
+            "slot_length_realtime": timedelta(0),
+            "start_date": date(2022, 10, 17),
+            "spot_market_type": 2,
+            "advanced_settings": None,
+            "random_seed": 0,
+            "capacity_kW": 5.0,
+            "grid_fee_type": 1,
+            "external_connection_enabled": False,
+            "currency": 0,
+            "settlement_market_enabled": False,
+            "relative_std_from_forecast_percent": 10.0,
+            "bid_offer_match_algo": 1,
+            "scm_coefficient_algorithm": 1,
+            "type": 0
+        }
+
+    @staticmethod
+    def _assert_all_settings_values(settings: Dict, compare_timedeltas: bool):
+        if compare_timedeltas:
+            assert settings["duration"] == timedelta(days=7)
+            assert settings["slot_length"] == timedelta(seconds=900)
+            assert settings["tick_length"] == timedelta(seconds=15)
+            assert settings["slot_length_realtime"] == timedelta(seconds=0)
+        else:
+            assert isclose(settings["duration"], timedelta(days=7).total_seconds())
+            assert isclose(settings["slot_length"], timedelta(seconds=900).total_seconds())
+            assert isclose(settings["tick_length"], timedelta(seconds=15).total_seconds())
+            assert isclose(settings["slot_length_realtime"], timedelta(seconds=0).total_seconds())
+        assert settings["market_count"] == 1
+        assert settings["cloud_coverage"] == 0
+        assert settings["pv_user_profile"] is None
+        assert settings["market_maker_rate"] == 30
+        assert settings["start_date"] == date(2022, 10, 17)
+        assert settings["spot_market_type"] == 2
+        assert settings["advanced_settings"] is None
+        assert settings["capacity_kW"] == 5.0
+        assert settings["grid_fee_type"] == 1
+        assert not settings["external_connection_enabled"]
+        assert settings["currency"] == 0
+        assert not settings["settlement_market_enabled"]
+        assert settings["relative_std_from_forecast_percent"] == 10.0
+        assert settings["bid_offer_match_algo"] == 1
+        assert settings["scm_coefficient_algorithm"] == 1
+        assert settings["type"] == 0
+
+    def test_simulation_settings_validator_works(self):
+        serializer = AVROSimulationSettingsSerializer()
```

Comment:
I suggest to do

```python
serializer = get_schema_validator(SOME_NAME)
```
here and in the following test methods.

Reply:
The main problem with this was that the Settings validator needed a different class than the normal validator. No problem at all though, I fixed it and changed the behavior of get_schema_validator to be similar to a factory method.